WordPress/Classes/Extensions/UIImageView+SiteIcon.swift

```diff
@@ -61,22 +61,60 @@ extension UIImageView {
     ///
     @objc
     func downloadSiteIcon(for blog: Blog, placeholderImage: UIImage? = .siteIconPlaceholder) {
+        func downloadImage(with request: URLRequest) {
+            self.downloadImage(
+                usingRequest: request,
+                placeholderImage: placeholderImage,
+                success: { [weak self] (image) in
+                    guard let self = self else {
+                        return
+                    }
+
+                    // In `MediaRequesAuthenticator.authenticatedRequestForPrivateAtomicSiteThroughPhoton` we're
+                    // having to replace photon URLs for Atomic Private Sites, with a call to the Atomic Media Proxy
+                    // endpoint.  The downside of calling that endpoint is that it doesn't always return images of
+                    // the requested size.
+                    //
+                    // The following lines of code ensure that we resize the image to the default Site Icon size, to
+                    // ensure there is no UI breakage due to having larger images set here.
+                    //
+                    let expectedSize = CGSize(width: SiteIconDefaults.imageSize, height: SiteIconDefaults.imageSize)
+
+                    if image.size != expectedSize {
+                        self.image = image.resizedImage(with: .scaleAspectFill, bounds: expectedSize, interpolationQuality: .default)
+                    } else {
+                        self.image = image
+                    }
+
+                    self.removePlaceholderBorder()
+            },
+                failure: { error -> () in
+                    if let error = error {
+                        CrashLogging.logError(error)
+                    }
+            })
+        }
```

Comment:
100000% sure this one needs to be a nested function? 

Just asking for sanity, i've not seen many of these around. Actually, first one in years!

Reply:
Need no, but I wasn't sure I wanted this method accessible from elsewhere.  The code around this logic needs a lot of cleanup.

I'm moving the logic to another method to resolve this comment.